WooCommerce/Classes/Authentication/Epilogue/StorePickerError.swift

```diff
@@ -1,21 +1,62 @@
 import SwiftUI
+import SafariServices
 
 /// Hosting controller wrapper for `StorePickerError`
 ///
 final class StorePickerErrorHostingController: UIHostingController<StorePickerError> {
+
+    /// Creates an `StorePickerErrorHostingController` with preconfigured button actions.
+    ///
+    static func createWithActions(presenting: UIViewController) -> StorePickerErrorHostingController {
+        let viewController = StorePickerErrorHostingController()
+        viewController.setActions(troubleshootingAction: {
+            let safariViewController = SFSafariViewController(url: WooConstants.URLs.troubleshootErrorLoadingData.asURL())
+            viewController.present(safariViewController, animated: true)
+        },
+        contactSupportAction: {
+            presenting.dismiss(animated: true) {
+                ZendeskManager.shared.showNewRequestIfPossible(from: presenting)
+            }
+        },
+        dismissAction: {
+            presenting.dismiss(animated: true)
+        })
+        return viewController
+    }
+
     init() {
         super.init(rootView: StorePickerError())
     }
 
     required dynamic init?(coder aDecoder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
-}
 
+    /// Actions are set in a separate function because most of the time, they will require to access `self` to be able to present new view controllers.
+    ///
+    func setActions(troubleshootingAction: @escaping () -> Void, contactSupportAction: @escaping () -> Void, dismissAction: @escaping () -> Void) {
+        self.rootView.troubleshootingAction = troubleshootingAction
+        self.rootView.contactSupportAction = contactSupportAction
+        self.rootView.dismissAction = dismissAction
+    }
+}
 
 /// Generic Store Picker error view that allows the user to contact support.
 ///
 struct StorePickerError: View {
+
+    /// Closure invoked when the "Troubleshooting" button is pressed
+    ///
+    var troubleshootingAction: () -> Void = {}
+
+    /// Closure invoked when the "Contact Support" button is pressed
+    ///
+    var contactSupportAction: () -> Void = {}
+
+    /// Closure invoked when the "Back To Sites" button is pressed
+    ///
+    var dismissAction: () -> Void = {}
```

Comment:
I'm not sure if it's a good practice to set default actions to empty, as normally we want the actions to work. This can cause trouble if one forgets to update the variables. Maybe it's safer to require these for initializing the view?

Reply:
We can't really pass them **only** upon initialization because most of the time those closures will need to access `self`. You can however still use the `initWithClosures` as they would be synthesized by Xcode.

To mitigate your worry there is a factory method that creates the `HostingViewController` with the appropriate closures set.